lighthouse-core/scoring.js

```diff
@@ -13,30 +13,33 @@ class ReportScoring {
    * @return {number}
    */
   static arithmeticMean(items) {
-    const results = items.reduce((result, item) => {
-      const score = Number(item.score) || 0;
-      const weight = Number(item.weight) || 0;
-      return {
-        weight: result.weight + weight,
-        sum: result.sum + score * weight,
-      };
-    }, {weight: 0, sum: 0});
+    const results = items.reduce(
+      (result, item) => {
+        const score = Number(item.score) || 0;
+        const weight = Number(item.weight) || 0;
+        return {
+          weight: result.weight + weight,
+          sum: result.sum + score * weight,
+        };
+      },
+      {weight: 0, sum: 0}
+    );
 
-    return (results.sum / results.weight) || 0;
+    return results.sum / results.weight || 0;
   }
 
   /**
    * Returns the report JSON object with computed scores.
-   * @param {{categories: !Object<string, {id: string|undefined, weight: number|undefined, audits: !Array<{id: string, weight: number|undefined}>}>}} config
+   * @param {{categories: !Object<string, {id: string|undefined, weight: number|undefined, score: number|undefined, audits: !Array<{id: string, weight: number|undefined}>}>}} config
    * @param {!Object<{score: ?number|boolean|undefined}>} resultsByAuditId
-   * @return {{score: number, categories: !Array<{audits: !Array<{score: number, result: !Object}>}>}}
+   * @void
    */
   static scoreAllCategories(config, resultsByAuditId) {
-    const categories = Object.keys(config.categories).map(categoryId => {
+    Object.keys(config.categories).forEach(categoryId => {
       const category = config.categories[categoryId];
       category.id = categoryId;
 
-      const audits = category.audits.map(audit => {
+      category.audits.forEach(audit => {
         const result = resultsByAuditId[audit.id];
         // Cast to number to catch `null` and undefined when audits error
```

Comment:
should all of these line just be moved into `audit.js` then to keep all the score mapping in the same place?

Reply:
more a comment for #4690 I suppose

Reply:
Aye. It actually already happens in that PR: https://github.com/GoogleChrome/lighthouse/pull/4690/files#diff-488f5da33160a651526463bdb951e5deL31